pkg/endpoint/bpf.go

```diff
@@ -416,6 +416,11 @@ func updateCT(owner Owner, e *Endpoint, epIPs []net.IP,
 // Must be called with endpoint.BuildMutex held.
 func (e *Endpoint) addNewRedirectsFromMap(owner Owner, m policy.L4PolicyMap, desiredRedirects map[string]bool) error {
 	for k, l4 := range m {
+		// Mark redirect as desired regardless of whether redirect has
+		// already been installed or not.
+		proxyID := e.ProxyID(&l4)
+		desiredRedirects[proxyID] = true
+
 		redirect := uint16(l4.L7RedirectPort)
 		if l4.IsRedirect() && redirect == 0 {
```

Comment:
I think the code should be:
```
redirect := uint16(l4.L7RedirectPort)
if l4.IsRedirect() {
	proxyID := e.ProxyID(&l4)
	desiredRedirects[proxyID] = true
	if redirect == 0 {
		...
	}
}
```
This makes sense to me. @tgraf, does that make sense to you?

Reply:
Yes, that makes sense to me. I'm not sure I understand the conditional on the redirect port